Build failure: _7zz

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *_7zz* with 5dceb649bf9db2d1217ce1d9ffd448f94efe556d

### Build log

```
       > cc      -O2 -c -Werror -Wall -Wextra -Waddress -Waggressive-loop-optimizations -Wattributes -Wcast-align -Wcomment -Wdiv-by-zero -Wformat-contains-nul -Winit-self -Wint-to-pointer-cast -Wunused -Wunused-macros  -Wbool-compare -Wduplicated-cond  -Waddress-of-packed-member -Wbool-operation -Wcast-align=strict -Wconversion -Wdangling-else -Wduplicated-branches -Wimplicit-fallthrough=5 -Wint-in-bool-context -Wmaybe-uninitialized -Wmisleading-indentation -Wmissing-attributes -DNDEBUG -D_REENTRANT -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -fPIC   -o b/g_x64/Aes.o ../../../../C/Aes.c
       > /nix/store/mm0pa3z7kk6jh1i9rkxqxjqmd8h1qpxf-bash-5.2p37/bin/bash: line 1: asmc: command not found
       > asmc -nologo -elf64 -DABI_LINUX -Fob/g_x64/ ../../../../Asm/x86/AesOpt.asm
       > make: *** [../../7zip_gcc.mak:1310: b/g_x64/LzmaDecOpt.o] Error 127
       > /nix/store/mm0pa3z7kk6jh1i9rkxqxjqmd8h1qpxf-bash-5.2p37/bin/bash: line 1: asmc: command not found
       > make: *** Waiting for unfinished jobs....
       > make: *** [../../7zip_gcc.mak:1265: b/g_x64/7zCrcOpt.o] Error 127
       > /nix/store/mm0pa3z7kk6jh1i9rkxqxjqmd8h1qpxf-bash-5.2p37/bin/bash: line 1: asmc: command not found
       > make: *** [../../7zip_gcc.mak:1291: b/g_x64/AesOpt.o] Error 127
       For full logs, run 'nix log /nix/store/qqp9cdcplmnxkg1jm8xlvpsnz2kdqxhb-7zz-24.08.drv'.
```

### Additional context

https://github.com/NixOS/nixpkgs/commit/5dceb649bf9db2d1217ce1d9ffd448f94efe556d

doesn't build anymore by default unless overriden with useUasm enabled

### Notify maintainers

@emilazy 

### Metadata
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.5, NixOS, 24.11 (Vicuna), 24.11.20241101.e9964f6`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/nat/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/nat/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/nat/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/nat/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/dm2kxl74dhsqvcj65qdq470n66cabqm3-gnome-settings-daemon-47.1/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/k1iys4jifr01ql4bzafl6ny71qzw32wz-lix-2.91.1/share`
 - nixpkgs: `/nix/store/y46bvh20g4lxvwsj2qqh2av5h9fmj7y1-source`
```

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/archivers/7zz/default.nix
+++ b/pkgs/tools/archivers/7zz/default.nix
@@ -77,6 +77,7 @@ stdenv.mkDerivation (finalAttrs: {
       "CXX=${stdenv.cc.targetPrefix}c++"
     ]
     ++ lib.optionals useUasm [ "MY_ASM=uasm" ]
+    ++ lib.optionals (!useUasm && stdenv.hostPlatform.isx86) [ "USE_ASM=" ]
     # We need at minimum 10.13 here because of utimensat, however since
     # we need a bump anyway, let's set the same minimum version as the one in
     # aarch64-darwin so we don't need additional changes for it

